app/models/mixins/has_monitoring_manager_mixin.rb

```diff
@@ -1,21 +1,25 @@
 module HasMonitoringManagerMixin
   extend ActiveSupport::Concern
 
-  private
+  def build_monitoring_manager_with_params
+    monitoring_manager = build_monitoring_manager(:parent_manager  => self,
+                                                  :name            => "#{name} Monitoring Manager",
+                                                  :zone_id         => zone_id,
+                                                  :provider_region => provider_region)
+    monitoring_manager
+  end
 
-  def ensure_monitoring_manager
-    # monitoring_manager should be defined by child classes.
-    if try(:monitoring_manager_needed?)
-      build_monitoring_manager(:parent_manager => self)
-      monitoring_manager.name = "#{name} Monitoring Manager"
+  def endpoint_created(role)
+    if role == "prometheus_alerts" && monitoring_manager.nil?
+      monitoring_manager = build_monitoring_manager_with_params
+      monitoring_manager.save
     end
-    ensure_monitoring_manager_properties
   end
 
-  def ensure_monitoring_manager_properties
-    if monitoring_manager
-      monitoring_manager.zone_id = zone_id
-      monitoring_manager.provider_region = provider_region
+  def endpoint_destroyed(role)
+    if role == "prometheus_alerts" && monitoring_manager.present?
+      # TODO: if someone deletes the alerts endpoint and then quickly readds it they can end up without a manager.
```

Comment:
not sure this is a real risk.  I think the queued destroy will be targeted to a specific manager id, and not affect the new monitoring manager.

Reply:
Ah, even less of a risk now that the logic moved to `endpoint.rb`.